File: src2/AzureLBUpdater/AzureLbUpdater/LoadBalancerRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AzureLbUpdater
{
    public class LoadBalancerRequest
    {
        public string Name { get; set; }
        public string Id { get; set; }
        public string Type { get; set; }
        public string Location { get; set; }

        public LoadBalancerProperties Properties { get; set; }

        public LoadBalancerRequest()
        {
            this.Properties = new LoadBalancerProperties();
        }

    }


    public class LoadBalancerProperties
    {
        public List<IpConfigurationRequest> IpConfigurations { get; set; }
        public VirtualMachineRequest VirtualMachine { get; set; }

        public LoadBalancerProperties()
        {
            this.VirtualMachine = new VirtualMachineRequest();
        }
    }

    public class IpConfigurationRequest
    {
        public string Name { get; set; }
        public string Id { get; set; }
        public IpConfigurationProperties Properties { get; set; }

        public IpConfigurationRequest()
        {
            this.Properties = new IpConfigurationProperties();
        }
    }


    public class IpConfigurationProperties
    {
        public SubnetRequest Subnet { get; set; }

        /// <summary>
        /// just a collection of resource ids
        /// </summary>
        public List<LoadBalancerBackendProperties> LoadBalancerBackendAddressPools { get; set; }

        public IpConfigurationProperties()
        {
            this.LoadBalancerBackendAddressPools = new List<LoadBalancerBackendProperties>();
        }
    }

    public class LoadBalancerBackendProperties
    {
        public string Id { get; set; }
    }

    public class SubnetRequest
    {
        public string Id { get; set; }
    }

    public class VirtualMachineRequest
    {
        public string Id { get; set; }
    }


}

File: src/LoadBalancerRequest.py
import json


class LoadBalancerRequest(object):
    """description of class"""
    def __init__(self, **kwargs):
        self.name = kwargs.get('name')
        #self.id = kwargs.get('id')
        #self.location = kwargs.get('location')
        #self.type = kwargs.get('type')
        #self.properties = kwargs.get('properties')

    def to_JSON(self):
        return json.dumps(self, default=lambda o: o.__dict__, 
            sort_keys=True, indent=4)

    def getName(self):
        return self.__name

    def setName(self, value):
        self.__name = value

    name = property(getName, setName)
    #@property
    #def name(self):
    #    return self.name

    #@name.setter
    #def name(self, value):
    #    self.name = value

    #@property
    #def id(self):
    #    return self.id

    #@id.setter
    #def id(self, value):
    #    self.id = value

    #@property
    #def type(self):
    #    return self.type

    #@type.setter
    #def type(self, value):
    #    self.type = value

    #@property
    #def location(self):
    #    return self.location

    #@location.setter
    #def location(self, value):
    #    self.location = value

    #@property
    #def properties(self):
    #    return self.properties

    #@properties.setter
    #def properties(self, value):
    #    self.properties = value

class LoadBalancerProperties(object):
    """description of class"""
    def __init__(self, **kwargs):
        self.name = kwargs.get('name') #dummy

class IpConfigurationRequest(object):
    def __init__(self, **kwargs):
        self.name = kwargs.get('name')  #dummy



class IpConfigurationProperties(object):
    def __init__(self, **kwargs):
        return super(IpConfigurationProperties, self).__init__(**kwargs)


class LoadBalancerBackendProperties(object):
    def __init__(self, **kwargs):
        return super(LoadBalancerBackendProperties, self).__init__(**kwargs)


class SubnetRequest(object):
    def __init__(self, **kwargs):
        return super(SubnetRequest, self).__init__(**kwargs)



class VirtualMachineRequest(object):
    def __init__(self, **kwargs):
        return super(VirtualMachineRequest, self).__init__(**kwargs)



